Route/RouteIdHandler.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Sonata\AdminBundle\Route;
+
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class RouteIdHandler
+{
+    /**
+     * @param Request $request
+     * @param AdminInterface $admin
+     *
+     * @return int|string
+     */
+    public function getObjectId(Request $request, AdminInterface $admin)
+    {
+        return $request->get($admin->getIdParameter());
+    }
+}
```

Comment:
Please configure your IDE to always add a newline at end of file, this will make the red pictogram disappear.